pitop/core/ImageFunctions.py

```diff
@@ -1,60 +1,42 @@
 from PIL import Image
-from numpy import array
+from numpy import (
+    asarray,
+    ndarray,
+)
 from urllib.request import urlopen
 
 from pitopcommon.formatting import is_url
 
 
-def __image_has_3_channels(image: array):
-    # RGB
-    return len(image.shape) == 3
-
-
-def __image_rgb_bgr_convert(image: array):
-    # For RGB - BGR conversion we can reverse the third level of the pixels ndarray
-    # https://stackoverflow.com/a/14140796
-    return image[:, :, ::-1]
-
-
-def pil_to_opencv(image: Image):
-    """
-    Converts an RGB Pillow Image into an OpenCV compatible BGR numpy ndarray.
-
-    :type image: PIL.Image.Image
-    :param image: A Pillow Image instance in raw RGB format
-    :rtype: numpy.ndarray
-    :return:
-        A numpy array representing the image in BGR format, as used by default in OpenCV
-    """
-    image = array(image)
-
-    if __image_has_3_channels(image):
-        # Array has 3 channel, do nothing
-        image = __image_rgb_bgr_convert(image)
-    else:
-        image = image
-
-    return image
-
-
-def opencv_to_pil(image: array):
-    """
-    Converts an OpenCV compatible BGR numpy ndarray into an RGB Pillow Image.
-
-    :type image: numpy.ndarray
-    :param image: Raw BGR image data as a numpy ndarray
-    :rtype: PIL.Image.Image
-    :return:
-        A Pillow Image in RGB format
-    """
-    image = array(image)
-    if __image_has_3_channels(image):
-        image = Image.fromarray(__image_rgb_bgr_convert(image))
-    else:
-        # Not 3 channel, do nothing
-        image = Image.fromarray(image)
-
-    return image
+def convert(image, format="PIL"):
+    try:
+        from cv2 import (
+            cvtColor,
+            COLOR_BGR2RGB,
+            COLOR_RGB2BGR,
+        )
+    except (ImportError, ModuleNotFoundError):
+        raise ModuleNotFoundError(
+            "OpenCV Python library is not installed. You can install it by running 'sudo apt install python3-opencv'.") from None
+
+    assert isinstance(format, str)
+    format = format.lower()
+    assert format in ("pil", "opencv")
```

Comment:
Seems like a nice generic format check function that could be used wherever `format=` is used?